IdentityCore/src/util/NSString+MSIDExtensions.m

```diff
@@ -424,4 +251,48 @@ - (BOOL)msidIsEquivalentWithAnyAlias:(NSArray<NSString *> *)aliases
     return NO;
 }
 
+
++ (NSString *)msidStringFromOrderedSet:(NSOrderedSet *)set
+{
+    NSInteger cSet = set.count;
+    if (cSet == 0)
+    {
+        return @"";
+    }
+    
+    NSMutableString *queryString = [[set objectAtIndex:0] mutableCopy];
+    
+    for (NSInteger i = 1; i < cSet; i++)
+    {
+        [queryString appendString:@" "];
+        [queryString appendString:[set objectAtIndex:i]];
+    }
+    
+    return queryString;
+}
+
+
+- (NSURL *)msidUrl
+{
+    return [[NSURL alloc] initWithString:self];
+}
+
+- (NSData *)msidData
+{
+    return [self dataUsingEncoding:NSUTF8StringEncoding];
+}
+
+
+- (NSOrderedSet<NSString *> *)msidScopeSet
+{
+    return [NSOrderedSet msidOrderedSetFromString:self];
+}
+
+
+- (NSString *)msidStringByRemovingPadding
```

Comment:
nit: I'd prefer this method to specify that it removes base64 padding, not any padding

Reply:
I actually got rid of this. it was only being used at one place and it is a one liner.